### PR TITLE
Added a client option that allows us to set the base url.

### PR DESCRIPTION
**Pull request by alesstimec**

### CODE REVIEW

File: api/plan/api_test.go [hunk 1]
```diff
@@ -117,9 +139,12 @@ func (s *clientSuite) TestAuthorize(c *gc.C) {
 type mockHttpClient struct {
```
Comment (tasdomas) [inline api/plan/api_test.go]: This should use Stub.

**Comment by cmars** [comment]
LGTM with Stub suggestion above.

File: api/plan/api_test.go (hunk 1 above)
Comment (alesstimec) [inline api/plan/api_test.go]: done.

**Comment by tasdomas** [comment]
LGTM

**Pull request merged**